### PR TITLE
Create a start repository on signup

### PR DESCRIPTION
**Pull request by SimonBarendse**
This sets up a repository and writes the first secret on signup,
so that the user can start by reading their first secret. This is
intended to smoothen onboarding of new users.

Also, we no longer print the email address should be validated first,
as we have moved this constraint to the moment when users start
collaborating (org invite).

### CODE REVIEW

File: internals/secrethub/signup.go [hunk 1]
```diff
@@ -158,6 +159,21 @@ func (cmd *SignUpCommand) Run() error {
...
+	fmt.Fprintln(cmd.io.Stdout(), "Setting up a start workspace.")
+
+	repoPath := secretpath.Join(cmd.username, "start")
+	_, err = client.Repos().Create(secretpath.Join(repoPath))
+	if err != nil {
+		return err
+	}
+
+	secretPath := secretpath.Join(repoPath, "hello")
+	_, err = client.Secrets().Write(secretPath, []byte("Welcome to SecretHub!"))
```
Comment (mackenbach) [inline internals/secrethub/signup.go]: Can we personalize this to use the user's `FullName`?

File: internals/secrethub/signup.go [hunk 2]
```diff
@@ -158,6 +159,21 @@ func (cmd *SignUpCommand) Run() error {
...
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(cmd.io.Stdout(), "Setup complete. You can read your first secret with `secrethub read %s`\n", secretPath)
```
Comment (mackenbach) [inline internals/secrethub/signup.go]: How about this message?

```
Setup complete. To read your first secret, run:

   secrethub read %s

```

File: internals/secrethub/signup.go (hunk 1 above)
Comment (mackenbach) [reply internals/secrethub/signup.go]: And no ASCII art?

Comment (mackenbach) [reply internals/secrethub/signup.go]: How about this message:

```
Welcome Simon Barendse! This is your first secret. To write a new version of this secret, run:

   secrethub write simon/start/hello

```

Then the first two steps of the guide are totally self explanatory. You don't even strictly require docs for this.

**Comment by mackenbach** [comment]
Also, how does this work for Dashboard signups that run `account init` (or was it `connect`)?

**Comment by SimonBarendse** [comment]
> Also, how does this work for Dashboard signups that run account init (or was it connect)?

`account init` is correct. I've added a commit so that a start repo is created there as well.

File: internals/secrethub/signup.go [hunk 3]
```diff
@@ -160,22 +162,28 @@ func (cmd *SignUpCommand) Run() error {
...
+	return createStartRepo(client, cmd.io.Stdout(), cmd.username, cmd.fullName)
+}
 
-	repoPath := secretpath.Join(cmd.username, "start")
-	_, err = client.Repos().Create(secretpath.Join(repoPath))
+// createStartRepo creates a start repository and write a fist secret to it, so that
+// the user can start by reading their first secret. This is intended to smoothen
+// onboarding.
+func createStartRepo(client secrethub.ClientInterface, w io.Writer, workspace string, name string) error {
+	fmt.Fprintln(w, "Setting up a start workspace...")
```
Comment (mackenbach) [inline internals/secrethub/signup.go]: What is a `start workspace`? Is `Setting up your workspace...` sufficient?

File: internals/secrethub/signup.go [hunk 4]
```diff
@@ -160,22 +162,28 @@ func (cmd *SignUpCommand) Run() error {
...
+	repoPath := secretpath.Join(workspace, "start")
+	_, err := client.Repos().Create(secretpath.Join(repoPath))
 	if err != nil {
 		return err
 	}
 
 	secretPath := secretpath.Join(repoPath, "hello")
-	message := fmt.Sprintf("Welcome %s! This is your first secret. To write a new version of this secret, run:\n\n    secrethub write %s", cmd.fullName, secretPath)
+	message := fmt.Sprintf("Welcome %s! This is your first secret. To write a new version of this secret, run:\n\n    secrethub write %s", name, secretPath)
```
Comment (mackenbach) [inline internals/secrethub/signup.go]: Spaces or tab(s) before the `secrethub write` text?

File: internals/secrethub/signup.go [hunk 5]
```diff
@@ -160,22 +162,28 @@ func (cmd *SignUpCommand) Run() error {
...
 
 	_, err = client.Secrets().Write(secretPath, []byte(message))
 	if err != nil {
 		return err
 	}
 
-	fmt.Fprintf(cmd.io.Stdout(), "Setup complete. To read your first secret, run:\n\n    secrethub read %s\n\n", secretPath)
+	fmt.Fprintf(w, "Setup complete. To read your first secret, run:\n\n    secrethub read %s\n\n", secretPath)
```
Comment (mackenbach) [inline internals/secrethub/signup.go]: Spaces or tab(s) before the `secrethub read` text?

File: internals/secrethub/signup.go (hunk 4 above)
Comment (SimonBarendse) [reply internals/secrethub/signup.go]: I'd say spaces, as we use them in other places as well (e.g. root command help text).

**Comment by SimonBarendse** [comment]
Let's postpone merging this until email validation is merged server-side.

**Pull request merged**